doc/efa-env-var.md

```diff
@@ -0,0 +1,105 @@
+   </thead>
+   <tr>
+      <td><code>FI_EFA_USE_HUGE_PAGE=0</code></td>
+      <td>Set to 0 when you see <code>os.fork()</code> causes <code>OSError: Cannot allocate
+      memory</code>. Typically happen by multi-process PyTorch data loader. Disabling huge page
+      causes minor performance hit, but it's needed to prevent fork fails due to the operating
+      system running out of huge pages.</td>
+   </tr>
+   <tr>
+      <td><code>FI_EFA_FORK_SAFE=1</code></td>
+      <td>Not needed for kernel>=5.15. Still fine to set it though no effect. See
+      <a href=https://github.com/ofiwg/libfabric/pull/9112>ref</a>.</td>
+   </tr>
+   <tr>
+      <td><code>FI_EFA_USE_DEVICE_RDMA=1</code></td>
+      <td>Do not set for libfabric>=1.18.0 and aws-ofi-nccl>=1.7.0. It's not harmful to set it on
+      p4/p5 instances with the newer software, but you just don't have to set it.</td>
+   </tr>
+   <tr>
+      <td><code>FI_EFA_ENABLE_SHM_TRANSFER=1</code></td>
+      <td>Not needed. This is really a no-op, as aws-ofi-nccl defaults to enable SHMEM.</td>
+   </tr>
+   <tr>
+      <td><code>FI_PROVIDER=efa</code></td>
+      <td>Use for aws-ofi-nccl<=1.5.0 AND p4/p5 instances.</td>
+   </tr>
+   <tr>
+      <td><code>NCCL_PROTO=simple</code></td>
+      <td>Use for aws-ofi-nccl<=1.5.0 and p4/p5 instances.</td>
+   </tr>
+   <tr>
+      <td><code>NCCL_SOCKET_NTHREADS</code></td>
+      <td>Not applicable for EFA.</td>
+   </tr>
+   <tr>
+      <td><code>NCCL_NSOCKS_PERTHREAD</code></td>
+      <td>Not applicable for EFA.</td>
+   </tr>
+   <tr>
+      <td><code>NCCL_MIN_CHANNELS=xxx</code></td>
+      <td>Recommend to leave it out to use the default. For e.g., on p4d/p4de, the number of
+      channels should be 8, which is the minimum for a 4-NIC platform. The reduction message is
+      split by number of GPUs in the job, then the number of channels, so having more channels than
+      necessary causes smaller messages which causes EFA to be starved for data.</td>
+   </tr>
+   <tr>
+      <td><code>NCCL_BUFFSIZE=xxx</code></td>
+      <td>Recommend to leave it out to use the default.</td>
+   </tr>
+   <tr>
+      <td><code>RDMAV_FORK_SAFE=1</code></td>
+      <td>Do not use. This is a RDMA-core environment variable. Prefer <code>FI_EFA_FORK_SAFE</code>
+      (if it still makes sense for your Linux kernel version). The two look the same, but actually
+      behaves very differently, especially on newer kernels, where <code>RDMAV_FORK_SAFE=1</code>
+      can break things.</td>
+   </tr>
+   <tr>
+      <td><code>RDMAV_*</code></td>
+      <td>Do not use.</td>
+   </tr>
+   <tr>
+      <td>NCCL version</td>
+      <td>Recommend one of the stable releases.</td>
+   </tr>
+</table>
+
+## 2. A word on p5.48xlarge instances
+
+Use cuda>=12.0, nccl>=2.18.0 (recommend at least 2.18.5), aws-ofi-nccl>=1.7.2 (recommend at least
+1.7.3).
+
+## 3. Sample Presets
+
+### 3.1. libfabric>=1.18.0 and aws-ofi-nccl>=1.7.0
+
+```bash
+export FI_EFA_USE_HUGE_PAGE=0
```

Comment:
Not sure about this. Are we recommending the user to set this always, or only when they run into the "cannot allocate memory" error?

Reply:
Only when running into the cannot allocate memory error.

Reply:
+1 to recommend disabling huge-page only when mem error arises.